packages/core-utils/src/route.ts

```diff
@@ -350,15 +362,23 @@ export function makeStringValueComparator(
 }
 
 /**
- * OpenTripPlanner sets the routeSortOrder to -999 by default. So, if that value
- * is encountered, assume that it actually means that the routeSortOrder is not
- * set in the GTFS.
+ * OTP1 sets the routeSortOrder to -999 by default. If we're encountering that value in OTP1,
+ * assume that it actually means that the route sortOrder is not set in the GTFS. If we encounter
+ * it in OTP2, it's a valid value, so we should return it.
  *
  * See https://github.com/opentripplanner/OpenTripPlanner/issues/2938
  * Also see https://github.com/opentripplanner/otp-react-redux/issues/122
+ * This was updated in OTP2 TO be empty by default. https://docs.opentripplanner.org/en/v2.3.0/OTP2-MigrationGuide/#:~:text=the%20Alerts-,Changes%20to%20the%20Index%20API,-Error%20handling%20is
  */
-function getRouteSortOrderValue(val: number): number {
-  return val === -999 ? undefined : val;
+export function getRouteSortOrderValue(route: Route): number {
+  const isOTP1 = !!route.agencyId;
```

Comment:
This line gives me pause because I know we are sometimes converting OTP2 data into OTP1 data but I think that's ok because in that case the behavior would be unchanged from the way it is now.